### PR TITLE
Includes has_and_belongs_to_many attributes in wrap_parameters 

### LINKED ISSUE #32803: wrap_parameters drops attributes for has_and_belongs_to_many association
When posting a flat JSON request to a controller which has wrap_parameters enabled for an associated model, the key-value for any has_and_belongs_to_many association is not present in the wrapped parameters.

### Steps to reproduce

```ruby
class User < ActiveRecord::Base
  has_and_belongs_to_many :roles
end

class UsersController < ActionController::Base
  wrap_parameters User, format: [:json]
end
```

The role_ids **are not** present when posting the user data in a **flat** JSON request, such as:

```json
{
  "name":     "Doe",
  "role_ids": [1, 2]
}
```

### Actual behavior

The actual params in the controller:

```ruby
{ "name" => "Doe", "role_ids" => [1, 2], "user" => { "name" => "Doe" } }
```

Note how the wrapped "user" params do not include the role_ids key.

### Expected behavior

The expected params in the controller:

```ruby
{ "name" => "Doe", "role_ids" => [1, 2], "user" => { "name" => "Doe", "role_ids" => [1, 2] } }
```

### System configuration

**Rails version**: <= v5.2.0

**Ruby version**: 2.5.0

### PR DESCRIPTION
**Pull request by eikes**
Previously only column attributes, stored_attributes and nested_attributes
were included in the wrapped parameters

Fixes #32803

### CODE REVIEW

**Comment by oniofchaos** [comment]
Looks like it needs a changelog but otherwise 👍

**Comment by eikes** [comment]
Is the changelog something I can write? @oniofchaos

**Comment by eikes** [comment]
Wtf?

File: actionpack/lib/action_controller/metal/params_wrapper.rb [hunk 1]
```diff
@@ -119,6 +119,14 @@ def include
                 end
               end
 
+              if m.respond_to?(:reflections) && m.reflections.keys.any?
```
Comment (rafaelfranca) [inline actionpack/lib/action_controller/metal/params_wrapper.rb:122]: Do we need to check the keys? Doesn't `m.reflections.any?` work?

File: actionpack/lib/action_controller/metal/params_wrapper.rb [hunk 2]
```diff
@@ -119,6 +119,14 @@ def include
...
+                self.include += m.reflections.select do |key, reflection|
+                  reflection.class == ActiveRecord::Reflection::HasAndBelongsToManyReflection
```
Comment (rafaelfranca) [inline actionpack/lib/action_controller/metal/params_wrapper.rb:124]: Why `has_and_belongs_to_many` is special? Should not `has_many through:` also be included here?

File: actionpack/test/controller/params_wrapper_test.rb [hunk 3]
```diff
@@ -269,6 +269,27 @@ def User.nested_attributes_options
...
+
+  def test_derived_wrapped_keys_from_has_and_belongs_to_many_association
+    require "active_record/reflection"
+
+    # User has_and_belongs_to_many :roles
+    def User.reflections
+      active_record_mock = MiniTest::Mock.new
+      active_record_mock.expect(:pluralize_table_names, true)
+      {
+        roles: ActiveRecord::Reflection::HasAndBelongsToManyReflection.new(:roles, {}, {}, active_record_mock)
```
Comment (rafaelfranca) [inline actionpack/test/controller/params_wrapper_test.rb:281]: Having to require this class is a smell that this should not be in Action Pack. Maybe we should check of something the reflection has, not which reflection it is?

**Pull request closed without merging**